### ISSUE TITLE
[Yacht] Remove legacy Python 2 Stub Comment

### ISSUE THREAD

**Issue by mohmad-null** [exercism/python#2827]
A number of exercises seem to have things explicitly working around Python 2 limitations. For example, anything that's using Enums.
Given Python 2 has been out of support for over 18 months, I'd suggest it's worth remove all of the Python-2-workaround stuff to follow best-practices that use a recent supported version of Python 3 (as of 3 days ago, this became 3.7).

For example: Yacht - https://github.com/exercism/python/blob/main/exercises/practice/yacht/yacht.py

**Comment by BethanyG** [comment]
Hi @mohmad-null, 

While the `Yacht` stub does indeed have a note about Python 2, using 
`NAME = some_value_expected_to_be_treated_as_constant` 
is not limited to Python 2, nor is the use of `Enum`  in place of that convention considered any more or less idiomatic in `Python 3.x`.

This track ceased support for Python 2.x somewhere in 2018 (if not before) -- though we may have missed  the odd comment or piece of example code.  Should you see additional examples of specific Python 2.x direction or code, please log them separately as bugs against a specific exercise, so they can be dealt with.  Thanks!

**Comment by mohmad-null** [comment]
Well, I was thinking that with 3(.4), you've got the [Enum Module](https://docs.python.org/3/library/enum.html). Given the purpose of Exercism is to promote/encourage learning, to me at least it seems like a good opportunity to use that.

The other obvious thing that's... well not 2.x specific, but now in all supported versions of Python 3.x is type hints. But that'll be a separate ticket. :-)

**Comment by BethanyG** [comment]
_No one is telling you that you **can't** use an `Enum` in your solution._  😉 .  We don't _prevent_ you -- should you see a need -- here or in any other exercise.  We're only requiring that you pass the given tests.  

Now, we do heavily bias you in the current stub  comment (_which we'll fix shortly_), and we do prompt with constant names -- so I may remove both, and see if having someone start from scratch results in different solutions.  But I suspect having the constants there might help some students to head in a productive direction.

So the tests don't _force_  or _require_ the use of an `Enum` type through explicit checks, and I'd prefer to keep it that way for several reasons:

1.  Tests for practice exercises should be closer to what might happen in unit tests for "production" or "real world" scenarios.  While we test (or try to) implementation for _concept_ exercises, we test **_results_** for practice exercises.  We're not interested in _how_ you make the result happen (_that's between you, your learning, and your mentor/teacher_) -- we are interested in your code producing an **_expected result_**.  

2. Enums are not currently a core  or idiomatic construct in Python -- yes -- they are very very useful in specific scenarios, but they're by no means "required" to be fluent in the language.  They're certainly not on the level of something like `with` or comprehensions, or generator expressions, or anything on the [`built-ins`](https://docs.python.org/3/library/functions.html) list .

3. Changing the tests to _require_ an `Enum` _type_ will invalidate [1,560 community solutions](https://exercism.org/tracks/python/exercises/yacht/solutions?passed_head_tests=true) without a very good reason.

We do have a [concept exercise requiring `Enums`](https://github.com/exercism/python/tree/main/exercises/concept/log-levels), but we have to publish it's prerequisites before we can set it live on the site.  I think that publishing that might help fill in the gap somewhat.

I also feel like a better solution to the lack of exercises that practice `Enums` might be _adding_ new practice exercises that better set up a scenarios where you would use one.  I am not sure that "backporting" existing exercises that were made prior to widespread `Enum` use is the correct way to go.

**Comment by BethanyG** [comment]
As for typehints .... I will warn you now that it won't happen anytime soon - if ever.  

Typehinting is considered **optional** in the reference implementation of Python for a very very good reason.  
**Python is dynamic and duck-typed.**  

If you want a static language, I'd recommend Go or Java or C#.  

Now, there are scenarios where typhinting is really good - but in those cases, it's usually paired with `mypy` for type enforcement.  If you aren't going to use `mypy` (_we don't_) to actually _check_ the type annotations and matches, they're no better than extra comments, and can also be inaccurate and contribute to visual noise.  This can be very confusing for learners, and really annoying for reviewers.

It's one of those things that is useful if you are going to be using it as a member of a dev team, and "extra" if you aren't going to.  While it can be helpful to learn -- I think that learning is best left to much later -- after a student has many more core concepts under their belt.

We do have typhinting on our roadmap for concept exercises, and I am thinking through what introducing `mypy` might look like, but neither is likely to happen in the first half of 2022.
